src/ray/common/task/task_spec.cc

```diff
@@ -38,7 +40,7 @@ SchedulingClass TaskSpecification::GetSchedulingClass(const ResourceSet &sched_c
                      << " types of tasks seen, this may reduce performance.";
     }
     sched_cls_to_id_[sched_cls] = sched_cls_id;
-    sched_id_to_cls_[sched_cls_id] = sched_cls;
+    sched_id_to_cls_.emplace(sched_cls_id, sched_cls);
```

Comment:
any reason to change this to emplace? If `sched_cls_id` exists, it won't update the value with emplace.

Reply:
Hmmm I think we need `emplace` because there's no longer a default empty constructor for the SchedulingClassDescriptor. All this logic is inside an if statement that should ensure there isn't an existing key/value.